src/usr/local/www/widgets/widgets/system_information.widget.php

```diff
@@ -30,27 +30,323 @@
+			$data_template = "<strong>%s</strong>";
+			$args[] = htmlspecialchars($config['system']['hostname'] . "." . $config['system']['domain']);
+			break;
+
+
+		case 'identifiers':
+			$data_template = "%s%s";
+			$args[] = sprintf('%s: <strong>%s</strong>', gettext("Serial"), system_get_serial());
+			$idfile = "/var/db/uniqueid";
+			$args[] = (file_exists($idfile) ? $args[] = sprintf("<br />\n%s: <strong>%s</strong>", gettext("Netgate Unique ID"), file_get_contents($idfile)) : "");
+			break;
+
+
+		case 'baseboard':
+			$_gb = exec('/bin/kenv -q smbios.bios.vendor 2>/dev/null', $biosvendor);
+			$_gb = exec('/bin/kenv -q smbios.bios.version 2>/dev/null', $biosversion);
+			$_gb = exec('/bin/kenv -q smbios.bios.reldate 2>/dev/null', $biosdate);
+			$_gb = exec('/bin/kenv -q smbios.planar.maker 2>/dev/null', $boardmaker);
+			$_gb = exec('/bin/kenv -q smbios.planar.product 2>/dev/null', $boardmodel);
+			$_gb = exec('/bin/kenv -q smbios.planar.version 2>/dev/null', $boardversion);
+			$_gb = exec('/bin/kenv -q smbios.planar.serial 2>/dev/null', $boardserial);
+
+			/* Only display information if there is any to show. */
+			if (strlen($boardmaker[0] . $boardmodel[0] . $boardserial[0]) != 0) {
+				$data_template = '%s<br/>%s%s';
+				$args[] = gettext('Vendor') . ': <strong>' . (empty($boardmaker[0]) ? gettext("Unknown") : $boardmaker[0]) . "</strong>";
+				$args[] = gettext('Model') . ': <strong>' . (empty($boardmodel[0]) ? gettext("Unknown") : $boardmodel[0] . (!empty($boardversion[0]) ? " ({$boardversion[0]})" : '')) . "</strong>";
+				$args[] = (empty($boardserial[0]) ? '' : '<br/>' . gettext('Serial') . ': <strong>' . $boardserial[0]  . "</strong>");
+			}
+			if (!empty($biosvendor[0]) || !empty($biosversion[0]) || !empty($biosdate[0])) {
+				$data_template .= '<br/>' . gettext('BIOS') . ":<div style='margin-left:20px'>%s<br/>\n%s</div>";
+				$args[] = gettext('Vendor') . ': <strong>' . (!empty($biosvendor[0]) ? "{$biosvendor[0]}" : "Unknown") . "</strong>";
+				$args[] = gettext('Version') . ': <strong>' . (!empty($biosversion[0]) ? $biosversion[0] : "Unknown") . (!empty($biosdate[0]) ? " ({$biosdate[0]})" : "") . "</strong>";
+			}
+			//  else nothing shown
+			break;
+
+
+		case 'firmware':
+			$data_template = "%s<br/><strong>%s</strong> (%s)<br />\n%s\n%s\n%s";
+			$args[] = $g['product_name'];
+			$args[] = $g['product_version_string'];
+			$args[] = php_uname("m");
+			$args[] = gettext('built on') . ' '. file_get_contents("/etc/version.buildtime");
+			$args[] = (!$g['hideuname'] ? '<br /><span title="' . php_uname("a") . '">(' . php_uname("s") . ' ' . php_uname("r") . ")</span>" : "");
+			$args[] = (!isset($config['system']['firmware']['disablecheck']) ? "<br /><br /><div id='system-information-widget-updatestatus'>" . gettext("Obtaining update status ") . '<i class="fa fa-cog fa-spin"></i></div>' : "");
+			break;
+
+
+		case 'platform':
+			if (!$g['hideplatform']) {
+				$is_nano = ($g['platform'] == "nanobsd");
```

Comment:
Note: for master (where this PR is targeted) there is no need for any nanoBSD code. But for back-porting it will be wanted. So maybe it is easy to leave this here, commit it, backport it, then take it out in master with a later commit.